test/utils/AutoDiscoveryUtils-test.tsx

```diff
@@ -87,6 +87,20 @@ describe("AutoDiscoveryUtils", () => {
             expect(logger.error).toHaveBeenCalled();
         });
 
+        it("throws an error when homeserver config has fail error and recognised error string", () => {
```

Comment:
have these AutoDiscovery tests made it into the wrong PR? I'm struggling to see how they are related.

Reply:
I was trying to make the coverage happier, but the bar is too high